packages/control-state-mixin/control-state-mixin.ts

```diff
@@ -145,33 +110,29 @@ export const ControlStateMixin = <T extends LitBase>(base: T): LitControlState &
         }
       }
 
-      if (props.has('tabIndex') && this._tabIndexValue !== undefined) {
-        this.focusElement.tabIndex = this._tabIndexValue;
+      if (props.has('tabIndex')) {
+        const { tabIndex } = this;
+        const oldTabIndex = props.get('tabIndex') as number;
+        this.focusElement.tabIndex = tabIndex == null ? oldTabIndex : tabIndex;
       }
     }
 
-    protected _setFocused(focused: boolean) {
-      if (focused) {
-        this.setAttribute('focused', '');
-      } else {
-        this.removeAttribute('focused');
-      }
-
-      // focus-ring is true when the element was focused from the keyboard.
-      // Focus Ring [A11ycasts]: https://youtu.be/ilj2P5-5CjI
-      if (focused && this._tabPressed) {
-        this.setAttribute('focus-ring', '');
-      } else {
-        this.removeAttribute('focus-ring');
-      }
+    protected _autoFocus() {
+      super._autoFocus && super._autoFocus(); // eslint-disable-line no-unused-expressions
+      this.focusElement.focus();
     }
 
-    private _bodyKeydownListener(e: KeyboardEvent) {
-      this._tabPressed = e.keyCode === 9;
+    protected _onFocusin(event: FocusEvent) {
+      const path = event.composedPath();
+      if (path[0] === this) {
+        this._focus();
+      } else if (path.indexOf(this.focusElement) !== -1 && !this.disabled && super._onFocusin) {
+        super._onFocusin(event);
+      }
     }
 
-    private _bodyKeyupListener() {
-      this._tabPressed = false;
+    protected _setFocused(focused: boolean) {
+      super._setFocused && super._setFocused(focused); // eslint-disable-line no-unused-expressions
     }
```

Comment:
Is this really needed when it only calls super?

Reply:
The purpose of this method was to only check for `super` in one place.
I agree it looks a bit weird but IMO it's easier to keep it this way.

Reply:
Discussed in person and agreed to change so we don't have `eslint-disable` comments.